File: packages/expo-location/android/src/main/java/expo/modules/location/exceptions/LocationBackgroundUnauthorizedException.java
package expo.modules.location.exceptions;

import expo.modules.core.interfaces.CodedThrowable;
import expo.modules.core.errors.CodedException;

public class LocationBackgroundUnauthorizedException extends CodedException implements CodedThrowable {
  public LocationBackgroundUnauthorizedException() {
    super("Not authorized to use background location services.");
  }

  @Override
  public String getCode() {
    return "E_LOCATION_BACKGROUND_UNAUTHORIZED";
  }
}

File: packages/expo-location/android/src/main/java/expo/modules/location/exceptions/LocationBackgroundUnauthorizedException.kt
package expo.modules.location.exceptions

import expo.modules.core.errors.CodedException

class LocationBackgroundUnauthorizedException : CodedException(
    "Not authorized to use background location services."
) {
  override fun getCode(): String {
    return "E_LOCATION_BACKGROUND_UNAUTHORIZED"
  }
}
